doc/programming/parsing.rst

```diff
@@ -305,6 +305,42 @@ representation as a ``uint64``:
     :exec: printf 12345 | spicy-driver %INPUT
     :show-with: foo.spicy
 
+.. _attribute_requires:
+
+.. rubric:: Enforcing Expected Field Values
```

Comment:
I wonder whether we should also allow `&requires` attributes on units to enforce global unit constraints. The way we handle it eagerly on fields means that constraints can only use data of fields up to and including the field having the `&requires` attribute, so to check a global unit constraint one would now need to e.g., add it to the last field.

I.e., instead of forcing users to write this
```
type X = unit {
    a: uint8;
    b: uint8;
    c: uint8 &requires=(self.c*self.c == self.a*self.a + self.b*self.b);
};
```
we could allow this form which should be less error-prone.

```
type X = unit {
    &requires=(self.c*self.c == self.a*self.a + self.b*self.b); // Maybe even allow this in an arbitrary position?
    
    a: uint8;
    b: uint8;
    c: uint8;
};
```